xDripG5/Transmitter.swift

```diff
@@ -269,35 +306,80 @@ fileprivate extension PeripheralManager {
         } catch let error {
             throw TransmitterError.controlError("Error enabling notification: \(error)")
         }
+    }
 
-        let timeMessage: TransmitterTimeRxMessage
+    fileprivate func readTimeMessage() throws -> TransmitterTimeRxMessage {
         do {
-            timeMessage = try writeMessage(TransmitterTimeTxMessage(), for: .control)
+            return try writeMessage(TransmitterTimeTxMessage(), for: .control)
         } catch let error {
             throw TransmitterError.controlError("Error getting time: \(error)")
         }
+    }
+
+    fileprivate func sendCommand(_ command: Command, activationDate: Date) throws -> TransmitterRxMessage {
+        switch command {
+        case .startSensor(let date):
+            let startTime = UInt32(date.timeIntervalSince(activationDate))
+            let startTimeEpoch = UInt32(date.timeIntervalSince1970)
+
+            do {
+                return try writeMessage(SessionStartTxMessage(time: startTime, timeEpoch: startTimeEpoch), for: .control)
+            } catch let error {
+                throw TransmitterError.controlError("Error starting session: \(error)")
+            }
+        case .stopSensor(let date):
+            let stopTime = UInt32(date.timeIntervalSince(activationDate))
+
+            do {
+                return try writeMessage(SessionStopTxMessage(time: stopTime), for: .control)
+            } catch let error {
+                throw TransmitterError.controlError("Error stopping session: \(error)")
+            }
+        case .calibrateSensor(let glucose, let date):
+            let unit = HKUnit.milligramsPerDeciliter()
+            let glucoseValue = UInt16(glucose.doubleValue(for: unit).rounded())
+            let time = UInt32(date.timeIntervalSince(activationDate))
+
+            do {
+                return try writeMessage(CalibrateGlucoseTxMessage(time: time, glucose: glucoseValue), for: .control)
+            } catch let error {
+                throw TransmitterError.controlError("Error calibrating sensor: \(error)")
+            }
+        }
 
-        let activationDate = Date(timeIntervalSinceNow: -TimeInterval(timeMessage.currentTime))
+    }
 
+    fileprivate func readGlucose(timeMessage: TransmitterTimeRxMessage, activationDate: Date) throws -> Glucose {
```

Comment:
This method is an extension of `PeripheralManager`, which is focused on providing easy, safe  command-response API on top of Core Bluetooth. It doesn't make sense for this method to take `timeMessage` and `activationDate` arguments for the sole purpose of constructing a `Glucose` object.

`readGlucose() throws -> GlucoseRxMessage` seems more appropriate.

Reply:
Thanks. Changed, here and also for `readCalibrationData`. Are you happy for `sendCommand` to take an `activationDate`?